Resubmit more code fakefp16 mapping unification

yinghai opened the pull request:
Summary: ATT

Test Plan: unittests

Differential Revision: D19632251

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit d6fb45c4:


* **1/2** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=b565d9b35696259f9ab7d5dfd3814c82d40aaa84&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base b565d9b3 since Jan 29
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  
* **1/2** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d6fb45c4682cc1153e5bdc438de6ca864986441b).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4374986) caffe2_onnx_py2_gcc5_ubuntu16_04_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92868678) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92868678))
<details>
<summary>
<code>Jan 29 23:47:03 CRITICAL:root:Cannot load caffe2.python. Error: /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/caffe2_pybind11_state.so: undefined symbol: _ZN6caffe23opt20getFakeFp16OpMappingB5cxx11Ebb</code>
</summary>

```
Jan 29 23:47:02 Successfully installed pytest-sugar-0.9.2 termcolor-1.1.0 
Jan 29 23:47:02 WARNING: You are using pip version 19.3.1; however, version 20.0.2 is available. 
Jan 29 23:47:02 You should consider upgrading via the 'pip install --upgrade pip' command. 
est/executor_test.py --ignore /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/operator_test/matmul_op_test.py --ignore /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/operator_test/pack_ops_test.py --ignore /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/mkl/mkl_sbn_speed_test.py --ignore /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/trt/test_pt_onnx_trt.py /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python 
Jan 29 23:47:03 ============================= test session starts ============================== 
Jan 29 23:47:03 platform linux2 -- Python 2.7.12, pytest-4.6.6, py-1.8.0, pluggy-0.13.0 -- /usr/bin/python2 
Jan 29 23:47:03 cachedir: .pytest_cache 
Jan 29 23:47:03 rootdir: /var/lib/jenkins 
Jan 29 23:47:03 plugins: sugar-0.9.2, hypothesis-3.44.6 
Jan 29 23:47:03 collecting ... WARNING:root:This caffe2 python run does not have GPU support. Will run in CPU only mode. 
Jan 29 23:47:03 CRITICAL:root:Cannot load caffe2.python. Error: /var/lib/jenkins/.local/lib/python2.7/site-packages/caffe2/python/caffe2_pybind11_state.so: undefined symbol: _ZN6caffe23opt20getFakeFp16OpMappingB5cxx11Ebb 
Jan 29 23:47:04 collected 0 items / 1 errors 
Jan 29 23:47:04  
Jan 29 23:47:04 ==================================== ERRORS ==================================== 
Jan 29 23:47:04 ________________________ ERROR collecting test session _________________________ 
Jan 29 23:47:04 /usr/local/lib/python2.7/dist-packages/py/_path/common.py:377: in visit 
Jan 29 23:47:04     for x in Visitor(fil, rec, ignore, bf, sort).gen(self): 
Jan 29 23:47:04 /usr/local/lib/python2.7/dist-packages/py/_path/common.py:419: in gen 
Jan 29 23:47:04     if p.check(dir=1) and (rec is None or rec(p))]) 
Jan 29 23:47:04 /usr/local/lib/python2.7/dist-packages/_pytest/main.py:667: in _recurse 
Jan 29 23:47:04     ihook = self.gethookproxy(dirpath) 
```

</details>

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [caffe2_onnx_py3_6_clang7_ubuntu16_04_test](https://dr.pytorch.org/build-details.html?build_id=92870818) from Jan 29



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Resubmit more code fakefp16 mapping unification (#32798)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/32798

ATT

Test Plan: unittests

Reviewed By: amylittleyang

Differential Revision: D19632251

fbshipit-source-id: ae665d2f902c8f9ced62ce70003640d82429668e
